thavalon-webapp/src/components/gameComponents/playerBoard.tsx

```diff
@@ -62,7 +72,23 @@ export function PlayerBoard(): JSX.Element {
         switch (message.messageType) {
             case InboundMessageType.Snapshot:
                 const snapshot = message.data as Snapshot
+                setMe(snapshot.me);
+                // Get proposal order, then get the most recent major message.
+                // Finally, feed the last message in
                 handleGameMessage(snapshot.log[0]);
+                for (let i = snapshot.log.length - 1; i >= 0; i--) {
+                    const logMessage = snapshot.log[i];
+                    switch (logMessage.messageType) {
+                        case GameMessageType.NextProposal:
+                        case GameMessageType.CommenceVoting:
+                        case GameMessageType.MissionGoing:
+                        case GameMessageType.BeginAssassination:
+                            handleGameMessage(logMessage);
+                            i = -1;
+                            break;
+                    }
+                }
+                handleGameMessage(snapshot.log[snapshot.log.length - 1]);
```

Comment:
If we're going to process the last message in log no matter what, shouldn't i start at snapshot.log.length - 2? Also similarly we double process snapshot.log[0], once at start and once at end of loop, that alright?

Reply:
Fixed the starting at last index issue. We don't double processes log[0] though. It is guaranteed the loop will exit prior to that, since we'll always have one of those message types before getting to proposalOrder. I updated the loop to be clearer, but we'll never double process it.